packages/ruby/spec/http_response_spec.rb

```diff
@@ -0,0 +1,43 @@
+require "http_response"
+
+RSpec.describe HttpResponse do
+  describe "#body" do
+    it "returns an empty string when the body is empty" do
+      response = HttpResponse.from_parts(200, {}, nil)
+
+      expect(response.body).to eq ""
+    end
+
+    it "concatenates the body when it is StringIO" do
+      body = "BODY"
+      response = HttpResponse.from_parts(200, {}, StringIO.new(body))
+
+      expect(response.body).to eq body
+    end
+  end
+
+  describe "#content_length" do
+    it "returns 0 for a response with an empty body status" do
+      response = HttpResponse.from_parts(204, {}, nil)
```

Comment:
Can we add a header and body here to make sure it's the status that is causing this behavior?